guides/source/layout.html.erb

```diff
@@ -39,59 +45,83 @@
         <li class="more-info"><a href="https://github.com/rails/rails">Contribute on GitHub</a></li>
       </ul>
     </div>
-  </div>
-  <div id="header">
+  </nav>
+  <header id="page_header">
     <div class="wrapper clearfix">
-      <div class="header-logo"><a href="index.html" title="Return to home page">Guides.rubyonrails.org</a></div>
-      <ul class="nav">
-        <li><a class="nav-item" href="index.html">Home</a></li>
-        <li class="guides-index guides-index-large">
-          <a href="index.html" id="guidesMenu" class="guides-index-item nav-item">Guides Index</a>
-          <div id="guides" class="clearfix" style="display: none;">
-            <hr />
-            <dl class="guides-section-container">
-              <% documents_by_section.each do |section| %>
-                <div class="guides-section">
-                  <dt><%= section['name'] %></dt>
+      <nav id="feature_nav"> <%# we use nav as an element to also provide the aria role and improve semantic meaning. %>
+        <div class="header-logo">
+          <a href="index.html" title="Return to home page">Guides</a>
+          <span id="version_switcher">
```

Comment:
We probably need to provide a way to detect that we're doing a public-site build, and only include the version switcher in that case. If someone is looking at it locally, I think it's clearer to only show their current version?

Reply:
Since we've removed the upper right corner banner, would we want to show that version number here in lieu of the switcher? How would we detect between a development build (where we want to see it), vs a general local build, vs production build?

Reply:
Improved to be explicit in 7e40297a

Reply:
Don't think we need to show anything there if there are no versions to select between. If you're looking at it locally, you know what you built from.

Reply:
I took a stab at this and was not able to sort it out (passing an ENV variable I could get to generator, but not then check for that in the layout.html.erb. Just a bit beyond my ability to heal.)

Reply:
Hi all, this is a blocker for us until we get some advice on how to implement here. Is this a big enough issue to stop this going live, or can we tackle this after? We have a small list of silo'd issues we will work on once the bulk of the work is out the door, and can add this to the list. cc: @matthewd @carlosantoniodasilva

Reply:
I am not sure there's much of a reason to _hide_ it locally and only show in a "production" build (edgeguides/guides), that would only make it harder to work on it locally (if you're working on the design for instance, you need a way to _see_ it so you can style it properly, etc.), since you'd now need to know that there's this element that only shows up in production builds, and how to build production vs non-production output for instance.

If you're building with a version different than Edge, it'd show that version as the currently one selected (with [my latest change](https://github.com/jathayde/rails/pull/2)), so it'd be similar to the previous "banner". Now you just get the other versions listed there, something we previously didn't have, and when selected they will just redirect you to the respective guides/edgeguides, even locally. That seems fine.

Right now I don't think we have any differentiation of building local vs building production, we just generate a bunch of static HTML to an output folder. Adding an extra "production vs local" variable just for hiding the version switcher (or hiding the other versions in there) seems to complicate it more.